core/queryresultio/sparqljson/src/test/resources/sparqljson/rdfstar-extendedformat-rdf4j.srj

```diff
@@ -0,0 +1,41 @@
+{
```

Comment:
Does it make sense to also add a test for the statement-encoded-as-base64 variant of the backwards-compatible output? And speaking of which, maybe even any generic plain JSON-LD output, as the SPARQL* parser should also be able to read that, right?

Reply:
> Does it make sense to also add a test for the statement-encoded-as-base64 variant of the backwards-compatible output?

The base-64 encoding is something the _writer_ uses to transform a triple value to a URI, so that it can be added to a result format serialization without the need for special extensions. It doesn't really make sense to add tests for it at the parser end because the whole point of it is that the parser treats it like any other URI. 

Arguably we could have some tests in place for the writer to check that the encoding is properly applied when the selected format doesn't support RDF* extensions, but that's a little out of scope for what we're doing here I think. I may look at that in a followup though as I may want to change the logic there a little bit.

> And speaking of which, maybe even any generic plain JSON-LD output, as the SPARQL* parser should also be able to read that, right?

Note that what we're fixing here is specifically the parser/writer for the SPARQL _table results_ format. JSON-LD is a different format, a serialization for RDF graphs (rather than table results), and it has a separate parser/writer. Currently, RDF4J's JSON-LD parser/writer does not support any RDF* extensions - the only RDF serialization formats that have been extended are Turtle (Turtle*), TriG (TriG*), and the RDF4J Binary RDF format. For all other formats, you can configure the writer to either convert RDF* triples to "classic" reification, or to use the same base64 encoding trick.

Reply:
I've created a followup issue for more configurable writing behavior: https://github.com/eclipse/rdf4j/issues/2437 . Separate PR coming up soon